--- SWAP/MOD_meteo.f90
+++ SWAP/MOD_meteo.f90
@@ -260,13 +260,12 @@
          end if
          
          ! check if meteo data should be loaded (dynamic crop growth)
-         i = 1
-         do while (tend - cropstart(i) > 0.d0)
-            if (cropstart(i) < 1.d0) exit
+         do i = 1, ifnd
+            if (tend - cropstart(i) <= 0.0d0) exit
+            if (cropstart(i) < 1.0d0) exit
             if (tend + 0.1d0 > cropstart(i) .AND. tstart - 0.1d0 < cropend(i)) then
                if (croptype(i) == 2) fl_loadmeteodata = .TRUE.
             end if
-            i = i + 1
          end do
       end if  
     
